src/corona_hakab_model/manager.py

```diff
@@ -121,32 +111,35 @@ def step(self):
     def progress_tests_and_isolation(self, new_tests: List[PendingTestResult]):
         self.new_detected_daily = 0
         new_results = self.pending_test_results.advance()
-        for agent, test_result, _ in new_results:
-            if test_result:
-                if not self.ever_tested_positive_vector[agent.index]:
-                    # TODO: awful late night implementation, improve ASAP
-                    self.new_detected_daily += 1
 
-            agent.set_test_result(test_result)
+        if new_results:
+            agent_indices, test_results, _ = zip(*new_results)
+            agent_indices = list(agent_indices)  # can't access numpy array with a tuple of indices
+            self.agents_df.set_tests_results(agent_indices, test_results)
+            tests_taken_mask = np.array([res != DetectionResult.NOT_TAKEN for res in test_results])
+            self.new_detected_daily = np.count_nonzero(
+                tests_taken_mask & self.agents_df.ever_tested_positive[agent_indices])
```

Comment:
ever tested positive might not be new detection

Reply:
fixed, if our definition of new detected is ok